src/mc/mcsim.f03

```diff
@@ -129,6 +136,96 @@ pure function list_confinement()
+                                                                      wlc_UP(:,i),wlc_UP(:,j),&
+                                                                      wlc_VP(:,i),wlc_VP(:,j))
+                            else
+                                delInt = delInt + internucleosome_energy(wlc_RP(:,j),wlc_RP(:,i),&
+                                                                      wlc_UP(:,j),wlc_UP(:,i),&
+                                                                      wlc_VP(:,j),wlc_VP(:,i))
+                            endif
+                        else ! i in moved, j not in moved
+                            if (j > i) then 
+                                delInt = delInt + internucleosome_energy(wlc_RP(:,i),wlc_R(:,j),&
+                                                                     wlc_UP(:,i),wlc_U(:,j),&
+                                                                     wlc_VP(:,i),wlc_V(:,j))
+                            else
+                                delInt = delInt + internucleosome_energy(wlc_R(:,j),wlc_RP(:,i),&
+                                                                     wlc_U(:,j),wlc_UP(:,i),&
+                                                                     wlc_V(:,j),wlc_VP(:,i))
+                            endif
+                        endif
+                    else 
+                        if (j >= left .AND. j <= right) then ! i not in moved, j in moved
+                            if (j > i) then 
+                                delInt = delInt + internucleosome_energy(wlc_R(:,i),wlc_RP(:,j),&
+                                                                     wlc_U(:,i),wlc_UP(:,j),&
+                                                                     wlc_V(:,i),wlc_VP(:,j))
+                            else
+                                delInt = delInt + internucleosome_energy(wlc_RP(:,j),wlc_R(:,i),&
+                                                                     wlc_UP(:,j),wlc_U(:,i),&
+                                                                     wlc_VP(:,j),wlc_V(:,i))
+                            endif
+                        else ! i not in moved, j not in moved
+                            if (j > i ) then 
+                                delInt = delInt + internucleosome_energy(wlc_R(:,i),wlc_R(:,j),&
+                                                                     wlc_U(:,i),wlc_U(:,j),&
+                                                                     wlc_V(:,i),wlc_V(:,j))
+                            else
+                                delInt = delInt + internucleosome_energy(wlc_R(:,j),wlc_R(:,i),&
+                                                                     wlc_U(:,j),wlc_U(:,i),&
+                                                                     wlc_V(:,j),wlc_V(:,i))
+                            endif
+                        endif
+                    endif
+                enddo
```

Comment:
Would it be possible to modularize the nucleosome interaction into a different file?  I've tried to keep MSsim as rather simple as it is some ways the main part of the MC code and a lot of different users will be looking through it.

Reply:
Yes absolutely! i 100% intended to do it but just never got around to it. my bad!